src/shared/lib/ExtendedRouterStore.ts

```diff
@@ -35,16 +32,13 @@ export default class ExtendedRouterStore extends (RouterStore as FunctionConstru
             }
         });
 
```

Comment:
I attempted to use the URI library to handle construction of the hash, but couldn't find a simple way to do it.   Also, I realized that $.param handles the uri encoding of parameter values, as demonstrated in test.  So, at this point, the only thing we're handling ourselves is the building of the parameter map, which seems very generic to me.